torchtext/prototype/models/t5/model.py

```diff
@@ -1,14 +1,14 @@
 from dataclasses import dataclass
-from typing import Dict, List, Optional, Union, Callable
+from typing import Dict, List, Mapping, Optional, Union, Callable
 
 import torch
 import torch.nn as nn
 from torch import Tensor
 
-from .modules import T5Encoder, T5Decoder, T5LayerNorm
+from .modules import T5Encoder, T5Decoder
 
 
-@dataclass
+@dataclass(frozen=True)
```

Comment:
Freezing this as we probably don't want people to be able to overwrite configs and still try to use the model - much more likely to run into bugs that way.

Reply:
Freezing the config won't make it impossible to try a smaller model or modified architecture. It just means that once they instantiate the config and pass the config to the model, they won't be able to modify it. 

Example:
```
config = T5Config(encoder_only=True)
t5_model = T5Model(config=config)
t5_model.config.encoder_only = False  # Currently allowed; with freezing config, this would throw an error
```

Reply:
Just to follow up here, in the example you just showed, would it affect the model behavior if users did end up changing the config after instantiating the model? IIUC the config is only used during model instantiation anyways. That being said I don't see any issues with freezing the config.

Reply:
It wouldn't affect the model behavior, but it would throw an error saying "Config cannot be modified", which I think is what we want. It would be considered undefined behavior if someone e.g. instantiated a model without a decoder and then went back and changed the config to say that it did have a decoder.